aleph/views/bookmarks_api.py

```diff
@@ -1,51 +1,88 @@
 import logging
 from flask import Blueprint, request
-from aleph.views.util import jsonify, require
+
+from aleph.core import db
+from aleph.search import DatabaseQueryResult
+from aleph.views.util import jsonify, require, parse_request
+from aleph.views.serializers import BookmarkSerializer
+from aleph.model import Bookmark
 
 
 log = logging.getLogger(__name__)
 blueprint = Blueprint("bookmarks_api", __name__)
 
 
+@blueprint.route("/api/2/bookmarks", methods=["GET"])
+def index():
+    """Get a list of bookmarks created by the current user
+    ---
+    get:
+      summary: Get bookmarks
+      tags: [Bookmarks]
+      responses:
+        '200':
+          description: OK
+          content:
+            application/json
+              schema:
+                type: object
+    """
+    require(request.authz.logged_in)
+
+    query = Bookmark.query.filter_by(role_id=request.authz.id)
+    result = DatabaseQueryResult(request, query)
+    return BookmarkSerializer.jsonify_result(result)
+
+
 @blueprint.route("/api/2/bookmarks", methods=["POST"])
 def create():
-    """Creates a new bookmark for the user. This is currently a stub endpoint,
-    as the current version of the bookmarks features stores bookmarks client side.
+    """Bookmark one or more entities.
     ---
     post:
-      summary: Create bookmark
+      summary: Create bookmarks
+      tags: [Bookmarks]
       responses:
         '200':
+          description: OK
           content:
             application/json:
               schema:
                 type: object
-          description: OK
-      tags:
-        - Bookmarks
     """
-    require(request.authz.logged_in)
-    log.info("User [%s]: Created bookmark", request.authz.role.id)
-    return jsonify({}, 200)
+    require(request.authz.session_write)
+    data = parse_request("BookmarkBulkCreate")
+
+    bookmarks = [
+        Bookmark(entity_id=bookmark.get("entity_id"), role_id=request.authz.id)
+        for bookmark in data
+    ]
+    db.session.add_all(bookmarks)
+    db.session.commit()
+
+    serializer = BookmarkSerializer()
+    response = serializer.serialize_many(bookmarks)
+    return jsonify(response, status=201)
 
 
-@blueprint.route("/api/2/bookmarks/<bookmark_id>", methods=["DELETE"])
-def delete(bookmark_id):
-    """Delete a bookmark. This is currently a stub endpoint, as the current version
-    of the bookmarks features stores bookmarks client side.
+@blueprint.route("/api/2/bookmarks/<entity_id>", methods=["DELETE"])
+def delete(entity_id):
```

Comment:
This endpoint should handle the request gracefully when the given bookmark doesn’t exist. Returning a 404 status code would be fine as well, we can handle this case explicitly on the frontend.